homeassistant/components/watttime/config_flow.py

```diff
@@ -0,0 +1,165 @@
+    {
+        vol.Required(CONF_LOCATION_TYPE): vol.In(
+            [LOCATION_TYPE_HOME, LOCATION_TYPE_COORDINATES]
+        ),
+    }
+)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for WattTime."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize."""
+        self._client: Client | None = None
+        self._password: str | None = None
+        self._username: str | None = None
+
+    async def async_step_coordinates(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the coordinates step."""
+        if not user_input:
+            return self.async_show_form(
+                step_id="coordinates", data_schema=STEP_COORDINATES_DATA_SCHEMA
+            )
+
+        if TYPE_CHECKING:
+            assert self._client
+
+        unique_id = f"{user_input[CONF_LATITUDE]}, {user_input[CONF_LONGITUDE]}"
+        await self.async_set_unique_id(unique_id)
+        self._abort_if_unique_id_configured()
+
+        try:
+            grid_region = await self._client.emissions.async_get_grid_region(
+                user_input[CONF_LATITUDE], user_input[CONF_LONGITUDE]
+            )
+        except CoordinatesNotFoundError:
+            return self.async_show_form(
+                step_id="coordinates",
+                data_schema=STEP_COORDINATES_DATA_SCHEMA,
+                errors={CONF_LATITUDE: "unknown_coordinates"},
+            )
+        except Exception as err:  # pylint: disable=broad-except
+            LOGGER.exception("Unexpected exception while getting region: %s", err)
+            return self.async_show_form(
+                step_id="coordinates",
+                data_schema=STEP_COORDINATES_DATA_SCHEMA,
+                errors={"base": "unknown"},
+            )
+
+        return self.async_create_entry(
+            title=unique_id,
+            data={
+                CONF_USERNAME: self._username,
+                CONF_PASSWORD: self._password,
+                CONF_LATITUDE: user_input[CONF_LATITUDE],
+                CONF_LONGITUDE: user_input[CONF_LONGITUDE],
+                CONF_BALANCING_AUTHORITY: grid_region["name"],
+                CONF_BALANCING_AUTHORITY_ABBREV: grid_region["abbrev"],
+            },
+        )
+
+    async def async_step_location(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the "pick a location" step."""
+        if not user_input:
+            return self.async_show_form(
+                step_id="location", data_schema=STEP_LOCATION_DATA_SCHEMA
+            )
+
+        if user_input[CONF_LOCATION_TYPE] == LOCATION_TYPE_COORDINATES:
+            return self.async_show_form(
+                step_id="coordinates", data_schema=STEP_COORDINATES_DATA_SCHEMA
```

Comment:
Please don't move directly to another step's form. Please call the corresponding step method instead to change step.